sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/data_transfer_func.py

```diff
@@ -228,53 +226,48 @@ def import_data(
     :param outputs: Mapping of outputs data bindings used in the job, default will be an output port with key "sink"
     and type "mltable".
     :type outputs: dict
-    :param is_deterministic: Specify whether the command will return same output given same input.
-        If a command (component) is deterministic, when use it as a node/step in a pipeline,
-        it will reuse results from a previous submitted job in current workspace which has same inputs and settings.
-        In this case, this step will not use any compute resource.
-        Default to be True, specify is_deterministic=False if you would like to avoid such reuse behavior.
-    :type is_deterministic: bool
     :param task: task type in data transfer component, possible value is "copy_data".
     :type task: str
     """
     source = _build_source_sink(source)
     outputs = outputs or {"sink": Output(type=AssetTypes.MLTABLE)}
     # # job inputs can not be None
     # job_inputs = {k: v for k, v in job_inputs.items() if v is not None}
-    component_outputs, job_outputs = _parse_inputs_outputs(outputs, parse_func=_parse_output)
+    _, job_outputs = _parse_inputs_outputs(outputs, parse_func=_parse_output)
     component = kwargs.pop("component", None)
     if component is None:
-        if source.type == ExternalDataType.DATABASE:
-            component_id = DataTransferBuiltinComponentUri.IMPORT_DATABASE
+        if source and source.type == ExternalDataType.DATABASE:
+            component = DataTransferBuiltinComponentUri.IMPORT_DATABASE
         else:
-            component_id = DataTransferBuiltinComponentUri.IMPORT_FILE_SYSTEM
-        component = DataTransferImportComponent(
+            component = DataTransferBuiltinComponentUri.IMPORT_FILE_SYSTEM
+        data_transfer_import_obj = DataTransferImport(
+            component=component,
             name=name,
+            description=description,
             tags=tags,
             display_name=display_name,
+            experiment_name=experiment_name,
+            compute=compute,
+            source=source,
+            outputs=job_outputs,
+            task=task,
+            **kwargs,
+        )
+        data_transfer_import_obj._source = ComponentSource.BUILTIN
+    else:
+        data_transfer_import_obj = DataTransferImport(
+            component=component,
```

Comment:
avoid writing these args twice?

Reply:
datatransfer node obj can only update _source after creatation, if pass source in __init__, will pop it in base_node init__.

We need set _source here to distinguish whether it's a built-in or a remote registry component

Reply:
will add a flag to judge whether need to update __source here